samples/04.0-file-transfer/integrationtest/src/test/java/org/eclipse/dataspaceconnector/extension/sample/test/FileTransferSampleTest.java

```diff
@@ -0,0 +1,299 @@
+    private static final String DESTINATION_FILE_PATH = "samples/04.0-file-transfer/consumer/requested.test.txt";
+    private static final Duration TIMEOUT = Duration.ofSeconds(15);
+    private static final Duration POLL_INTERVAL = Duration.ofMillis(1000);
+    private static final String API_KEY_HEADER_KEY = "X-Api-Key";
+    private static final String API_KEY_HEADER_VALUE = "password";
+    @RegisterExtension
+    protected static EdcRuntimeExtension provider = new EdcRuntimeExtension(
+            ":samples:04.0-file-transfer:provider",
+            "provider",
+            Map.of(
+                    // Override 'edc.samples.04.asset.path' implicitly set via property 'edc.fs.config'.
+                    "edc.samples.04.asset.path", new File(TestUtils.findBuildRoot(), SAMPLE_ASSET_FILE_PATH).getAbsolutePath(),
+                    "edc.fs.config", new File(TestUtils.findBuildRoot(), PROVIDER_CONFIG_PROPERTIES_FILE_PATH).getAbsolutePath()
+            )
+    );
+    // Starting EDC runtimes implicitly aligns to Run the sample / 1. Build and start the connectors.
+    @RegisterExtension
+    protected static EdcRuntimeExtension consumer = new EdcRuntimeExtension(
+            ":samples:04.0-file-transfer:consumer",
+            "consumer",
+            Map.of(
+                    "edc.fs.config", new File(TestUtils.findBuildRoot(), CONSUMER_CONFIG_PROPERTIES_FILE_PATH).getAbsolutePath()
+            )
+    );
+    private final File destinationFile = new File(TestUtils.findBuildRoot(), FileTransferSampleTest.DESTINATION_FILE_PATH);
+    private String contractNegotiationId;
+    private String contractAgreementId;
+
+    private static Properties readPropertiesFile(String fileName) throws IOException {
+        FileInputStream fis = null;
+        Properties prop = new Properties();
+
+        try {
+            fis = new FileInputStream(fileName);
+            prop.load(fis);
+
+        } finally {
+            if (fis != null) fis.close();
+        }
+
+        return prop;
+    }
+
+    /**
+     * Run all sample steps in one single test.
+     * Note: Sample steps cannot be separated into single tests because {@link EdcRuntimeExtension}
+     * runs before each single test.
+     */
+    @Test
+    void runSampleSteps() throws IOException {
+        assertTestPrerequisites();
+
+        assertConfigPropertiesUniquePorts();
+        assertInitiateContractNegotiation();
+        assertLookUpContractAgreementId();
+        assertRequestFile();
+        assertWaitForDestinationFileExistence();
+
+        cleanTemporaryTestFiles();
+    }
+
+    /**
+     * Assert that prerequisites are fulfilled before running the test.
+     * This assertion checks only whether the file to be copied is not existing already.
+     */
+    private void assertTestPrerequisites() {
+        var transferredFile = new File(TestUtils.findBuildRoot(), DESTINATION_FILE_PATH);
+
+        Assertions.assertThat(transferredFile.exists()).isFalse();
```

Comment:
```suggestion
        assertThat(transferredFile).exists();
```
- static import
- domain specific assertions